src/bit_str.jl

```diff
@@ -83,11 +98,13 @@ for op in [:<, :>, :(<=), :(>=)]
     @eval Base.$op(a::T, b::T) where {T<:BitStr} = Base.$op(buffer(a), buffer(b))
 end
 
-for op in [:(==)]
-    @eval Base.$op(a::T, b::Number) where {T<:BitStr} = Base.$op(buffer(a), b)
-    @eval Base.$op(a::Number, b::T) where {T<:BitStr} = Base.$op(a, buffer(b))
-    @eval Base.$op(a::BitStr{N}, b::BitStr{N}) where {N} = Base.$op(buffer(a), buffer(b))
+for Type in [Number, BigInt, BigFloat, AbstractIrrational, Rational, Complex]
+    @eval Base.:(==)(a::T, b::$Type) where {T<:BitStr} = Base.:(==)(buffer(a), b)
+    @eval Base.:(==)(a::$Type, b::T) where {T<:BitStr} = Base.:(==)(a, buffer(b))
```

Comment:
Maybe we only want to compare `BitStr` and `Integer` types. e.g. It does not make sense to compare BitStr and complex numbers.